src/Cake.Tfs/PullRequest/TfsPullRequestSettings.cs

```diff
@@ -66,6 +66,44 @@ public TfsPullRequestSettings(TfsPullRequestSettings settings)
             this.ThrowExceptionIfPullRequestCouldNotBeFound = settings.ThrowExceptionIfPullRequestCouldNotBeFound;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TfsPullRequestSettings"/> class using environment variables.
+        /// </summary>
+        /// <param name="credentials">Credentials to use to authenticate against Team Foundation Server or
+        /// Azure DevOps.</param>
+        public TfsPullRequestSettings(ITfsCredentials credentials)
+        {
+            credentials.NotNull(nameof(credentials));
+            this.Credentials = credentials;
+
+            var repositoryUrl = Environment.GetEnvironmentVariable("BUILD_REPOSITORY_URI", EnvironmentVariableTarget.Process);
+            repositoryUrl.NotNullOrWhiteSpace(nameof(repositoryUrl));
```

Comment:
This will throw an `ArgumentNullException` which might be misleading as no argument is involved. It would be better to throw an `InvalidOperation` exception. It would also be helpful to point the user in the right direction as probably is not running in an Azure Pipelines build.

Reply:
Makes sense. Modified to use `InvalidOperation` exception.